src/objectregistry.h

```diff
@@ -56,10 +43,17 @@ class ObjectRegistry {
 
     static EventuallyPersistentEngine *onSwitchThread(EventuallyPersistentEngine *engine,
                                                       bool want_old_thread_local = false);
+};
 
-    static void setStats(std::atomic<size_t>* init_track);
-    static bool memoryAllocated(size_t mem);
-    static bool memoryDeallocated(size_t mem);
+class SystemAllocationGuard {
+public:
+    SystemAllocationGuard();
+    ~SystemAllocationGuard();
+private:
+    EventuallyPersistentEngine* engine = nullptr;
+    arenaid_t arenaId = 0;
 };
 
+#define __system_alloc__ SystemAllocationGuard __s_a_g__;
```

Comment:
- *__system_alloc__* had to be defined because the threadlocal engine* becomes stale when an engine is destroyed & to set the arena correctly we need the arenaId from the engine & I was not able to detect if the ptr was stale